src/main/java/careeratlas/backend/Service/JobSearchServiceImpl.java

```diff
@@ -1,34 +1,39 @@
 package careeratlas.backend.Service;
 
-import careeratlas.backend.Domain.JobResponse;
+import careeratlas.backend.Domain.IndeedResponse;
 import careeratlas.backend.Domain.JobSearch;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
+import java.io.IOException;
 import javax.ws.rs.client.*;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
-import java.util.HashMap;
 
 @Service
 public class JobSearchServiceImpl implements JobSearchService {
 
     @Override
-    public ArrayList<JobResponse> searchJobs(String jobTitle, String jobType, String distance, String location){
+    public IndeedResponse searchJobs(String jobTitle, String jobType, String distance, String location){
         JobSearch jobSearch = new JobSearch(location, jobTitle, distance, jobType);
         Client client = ClientBuilder.newClient();
         WebTarget webTarget = client.target(jobSearch.getUrl());
         Invocation.Builder invocationBuilder = webTarget.request(MediaType.APPLICATION_JSON);
         Response response = invocationBuilder.get();
 
-        ArrayList<HashMap> indeedSearchResults = ((ArrayList<HashMap>) response.readEntity(HashMap.class).get("results"));
-        ArrayList<JobResponse> parsedSearchResults = new ArrayList<JobResponse>();
-        for (HashMap var : indeedSearchResults)
-        {
-            JobResponse job = new JobResponse(var.get("company").toString(), var.get("jobtitle").toString(), var.get("jobkey").toString(), var.get("longitude").toString(), var.get("latitude").toString(), var.get("url").toString(), var.get("snippet").toString());
-            parsedSearchResults.add(job);
+        String indeedSearchResults = response.readEntity(String.class);
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        IndeedResponse indeedResponse = null;
+        try {
+            indeedResponse = mapper.readValue(indeedSearchResults, IndeedResponse.class);
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-        return parsedSearchResults;
+
+        return indeedResponse;
```

Comment:
This won't return a null ever, it will at least always have some basic info in the header but there might not be any jobs associated with a search. That will probably have to be handled on the frontend but this Optional is good to know

Reply:
If `readValue` throws an exception `indeedResponse` is null.

Reply:
Ok, so last quick question. With how basic my endpoint is right is it worth creating a whole Optional object instead of just a `indeedResponse == null` check instead?

Reply:
Totally your call. If anything, I would almost say dont catch the exception there if there is not a response that makes sense to return. Someone is going to call that method expecting an `IndeedResponse`. If you are unable to return that (it could be an 'error' `IndeedResponse`), it would make sense to let whoever cares catch the exception and handle in their scope. So your controller could catch the exception and return a 500 to the UI. All of these are just design decisions and since its just a personal project its basically down to your opinion.

Reply:
This can be handled in another PR, it works for now